packages/@aws-cdk/cdk/README.md

```diff
@@ -3,3 +3,149 @@
 This library includes the basic building blocks of 
 the [AWS Cloud Development Kit](https://github.com/awslabs/aws-cdk) (AWS CDK).
 
+## Aspects
+
+Aspects are a mechanism to extend the CDK without having to directly impact the
+class hierarchy. We have implemented Aspects using the [Visitor
+Pattern](https://en.wikipedia.org/wiki/Visitor_pattern).
+
+An aspect in the CDK is defined by this [interface](lib/aspects/aspect.ts)
+
+Aspects can be applied to any construct. During the tree
+preparation phase the aspect will visit each construct in the tree one
+time. Aspects are invoked in the order they were added to the construct,
+starting at the `App` (root of the tree) and progressing in order to the leaf
+nodes (most commonly the CloudFormation Resource). Aspect authors will implement 
+the `visit(IConstruct)` function and can inspect the `Construct` for specific characteristics.
+Such as, is this construct a CloudFormation Resource?
+
+## Tagging
+
+Tags are implemented using Aspects.
```

Comment:
Use "aspects" as a noun without capitalization. Once you defined it, no need to capitalize everywhere